src/arrayMethodSort.js

```diff
@@ -3,9 +3,46 @@
 /**
  * Implement method Sort
  */
+
+function compareAsString(a, b) {
+  const valueA = String(a);
+  const valueB = String(b);
+
+  if (valueA > valueB) {
+    return 1;
+  }
+
+  if (valueA < valueB) {
+    return -1;
+  }
+
+  return 0;
+}
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsString) {
+    let count;
+
+    if (compareFunction === undefined) {
+      return this;
+    }
+
+    do {
+      count = 0;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const a = this[i];
+        const b = this[i + 1];
```

Comment:
For readibility you could name them like:
```suggestion
        const current = this[i];
        const next = this[i + 1];
```